src/pipeline.js

```diff
@@ -94,6 +94,37 @@ class Pipeline {
     this._posts = [];
     // functions that are executed before each step
     this._taps = [];
+
+    this.attach = function attach(ext) {
+      if (this.sealed) {
+        return;
+      }
+      if (ext && ext.before && typeof ext.before === 'object') {
+        Object.keys(ext.before).map(key => this.before(ext.before[key], key));
+      }
+      if (ext && ext.after && typeof ext.after === 'object') {
+        Object.keys(ext.after).map(key => this.after(ext.after[key], key));
+      }
+      this.sealed = true;
+    };
+
+    this.attach.generic = (f, name, offset = 0) => {
+      const re = new RegExp(`^.*\\:${name} from `);
```

Comment:
I am afraid I am having a bit of trouble understanding this function. Could you add some comments to elaborate on how the implementation works?

Reply:
how does a developer _know_ the name of the functions?
after all, it is very volatile and defined by the `pipe` function used.